### PR TITLE
Added remove packaging folder at the start of stack package

### PR DESCRIPTION
**Pull request by CameronMcWilliam**
Even though we had a defer that removes the packaging folder where templating takes place.  If a user used CTRL-C after `stack package` it will not remove the packaging folder and would error on the next use, then remove it on the defer.  I have added a `removeAll` to the start of the function as to avoid this error.

### CODE REVIEW

File: cmd/stack_package.go [hunk 1]
```diff
@@ -146,6 +146,12 @@ The packaging process builds the stack image, generates the "tar.gz" archive fil
 				return errors.Errorf("Error creating stacks directory: %v", err)
 			}
 
+			err = os.RemoveAll(stackPath)
+
+			if err != nil {
+				return errors.Errorf("Error creating removing packaging directory: %v", err)
```
Comment (EnriqueL8) [inline cmd/stack_package.go]: error message typo

File: cmd/stack_package.go [hunk 2]
```diff
@@ -146,6 +146,12 @@ The packaging process builds the stack image, generates the "tar.gz" archive fil
 				return errors.Errorf("Error creating stacks directory: %v", err)
 			}
 
+			err = os.RemoveAll(stackPath)
```
Comment (EnriqueL8) [inline cmd/stack_package.go]: should we remove the one in line 158 `os.RemoveAll(stackPath)` and the `defer` in 163?

Comment (CameronMcWilliam) [reply cmd/stack_package.go]: Don't think so.  We shouldn't keep the packaging folder if possible.

File: cmd/stack_package.go [hunk 3]
```diff
@@ -673,3 +682,17 @@ func ApplyTemplating(stackPath string, templateMetadata interface{}) error {
 	return nil
 
 }
+
+func removePackagingFolder(stackPath string) error {
```
Comment (EnriqueL8) [inline cmd/stack_package.go]: This function looks good, thanks! Just a suggestion would be to rename it to `removePackagingFolderIfExists` or even rename it to `removeFolderIfExists` and move it to `utils.go`, but I will leave this to your judgement!

Comment (CameronMcWilliam) [reply cmd/stack_package.go]: I agree I like this more generic approach.  I will do that now.

**Pull request merged**